src/client_side.cc

```diff
@@ -988,6 +988,7 @@ ConnStateData::kick()
     } else {
         // XXX: Can this happen? CONNECT tunnels have deferredRequest set.
         debugs(33, DBG_IMPORTANT, MYNAME << "abandoning " << clientConnection);
+        clientConnection->close();
```

Comment:
My PR eliminates the only case I have found where the "abandoning" state is reached, so that part of the PR can be removed if it is of concern.

It was included because hitting the "abandoning" state is:
1. Likely to result in a socket being leaked, which will ultimately crash the server; and
2. Not hit during normal operation, in my experience.

I felt that, although the behaviour isn't characterised well enough to know for sure that legitimate requests will never end up in that state, it is preferable to drop the occasional legitimate request than to crash the server.

Reply:
Please do not misinterpret my question as me fishing for ammunition to undo this change. I was aware of your original motivation here (thank you for documenting it!) and just needed a bit more info to get the remaining code in sync with this change. Thank you for confirming that you no longer see these "abandoning" messages. I am resolving this high-level change request.